src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/FindAllReferencesHandler.cs

```diff
@@ -65,23 +67,8 @@ internal class FindAllReferencesHandler : IRequestHandler<ReferenceParams, VSRef
             _lspProgressListener = lspProgressListener;
         }
 
-        // Roslyn sends Progress Notifications every 0.5s *only* if results have been found.
-        // Consequently, at ~ time > 0.5s ~ after the last notification, we don't know whether Roslyn is
-        // done searching for results, or just hasn't found any additional results yet.
-        // To work around this, we wait for up to 3.5s since the last notification before timing out.
-        //
         // Internal for testing
-        internal TimeSpan WaitForProgressNotificationTimeout { private get; set; } = TimeSpan.FromSeconds(3.5);
-
-        public async Task<VSReferenceItem[]> HandleRequestAsync(ReferenceParams request, ClientCapabilities clientCapabilities, CancellationToken cancellationToken)
```

Comment:
Is this unrelated?

Reply:
I refactored out some `IProgress<T>` logic from FARHandler to a `LSPProgressListenerHandlerBase` to be shared with the `DocumentPullDiagnosticsHandler`.